client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js

```diff
@@ -69,155 +82,82 @@ class PaymentMethodStripe extends Component {
-						<NoticeAction href="https://dashboard.stripe.com/register">{ translate( 'Sign up' ) }</NoticeAction>
-					</Notice>
-					<FormLabel>{ translate( 'Payment Mode' ) }</FormLabel>
-					<SegmentedControl
-						primary
-					>
-						<ControlItem
-							selected={ method.settings.testmode.value === 'yes' }
-							onClick={ this.onToggleTestMode( 'test' ) }
-						>
-							{ translate( 'Test Mode' ) }
-						</ControlItem>
-
-						<ControlItem
-							selected={ method.settings.testmode.value === 'no' }
-							onClick={ this.onToggleTestMode( 'live' ) }
-						>
-							{ translate( 'Live Mode' ) }
-						</ControlItem>
-					</SegmentedControl>
-				</FormFieldset>
-				{ method.settings.testmode.value === 'yes' && this.renderKeyFields( false ) }
-				{ method.settings.testmode.value === 'no' && this.renderKeyFields( true ) }
-				<FormFieldset className="payments__method-edit-field-container">
-					<FormLegend>{ translate( 'Payment authorization' ) }</FormLegend>
-					<FormLabel>
-						<FormRadio
-							name="capture"
-							value="yes"
-							checked={ 'yes' === method.settings.capture.value }
-							onChange={ this.onEditFieldHandler } />
-						<span>{ translate( 'Authorize and charge the customers credit card automatically' ) }</span>
-					</FormLabel>
-					<FormLabel>
-						<FormRadio
-							name="capture"
-							value="no"
-							checked={ 'no' === method.settings.capture.value }
-							onChange={ this.onEditFieldHandler } />
-						<span>{ translate( 'Authorize the customer\'s credit card but charge manually' ) }</span>
-					</FormLabel>
-				</FormFieldset>
-				<FormFieldset>
-					<FormLabel>
-						{ translate( 'Descriptor' ) }
-					</FormLabel>
-					<FormTextInput
-						name="statement_descriptor"
-						onChange={ this.onEditFieldHandler }
-						value={ method.settings.statement_descriptor.value }
-						placeholder={ this.getStatementDescriptorPlaceholder() } />
-					<FormSettingExplanation>
-						{ translate( 'Appears on your customer\'s credit card statement. 22 characters maximum' ) }
-					</FormSettingExplanation>
-				</FormFieldset>
-				<FormFieldset className="payments__method-edit-field-container">
-					<FormLabel>
-						{ translate( 'Use Apple Pay' ) }
-					</FormLabel>
-					<PaymentMethodEditFormToggle
-						checked={ method.settings.apple_pay.value === 'yes' ? true : false }
-						name="apple_pay"
-						onChange={ this.onEditFieldHandler } />
-					<span>
-						{ translate(
-							'By using Apple Pay you agree to Stripe and ' +
-							'Apple\'s terms of service'
-						) }
-					</span>
-				</FormFieldset>
-			</Dialog>
+			<PaymentMethodStripeKeyBasedDialog
+				method={ method }
+				onCancel={ onCancel }
+				onDone={ onDone }
+				onEditField={ this.onEditFieldHandler }
+				onUserRequestsConnectFlow={ connectFlowsEnabled ? this.onUserRequestsConnectFlow : undefined }
+				site={ site }
```

Comment:
I kind of reviewed this backwards, bottom diffs to top 😆  - so it seemed - though i could be wrong - that the `site.domain` is what is used in the Dialogs. If so would it be simpler to just pass the domain along here? I might have missed some other uses of the`site` though so sorry if I did.

Reply:
Fixed in https://github.com/Automattic/wp-calypso/pull/17197/commits/4b5f156ddc505a0e4a4f4271a7531f008651485e